### PR TITLE
[2113, 2303]: restricting actions and pending tasks viewed for court room manager

### PR DESCRIPTION
**Pull request by ashishbeehyv**
https://github.com/pucardotorg/dristi/issues/2113
https://github.com/pucardotorg/dristi/issues/2303

## Summary by CodeRabbit

- **New Features**
	- Introduced role-based visibility for UI components, specifically for users with the "COURT_ROOM_MANAGER" role.
	- Action bar and "Take Action" button are now hidden for courtroom managers.
	- Form behavior is adjusted for courtroom managers, disabling certain features and modifying visibility of labels.
	- Pending tasks are filtered out for courtroom managers, ensuring they only see relevant tasks.
	- Enhanced access control in modals and dropdowns based on user roles, improving user experience.
	- Conditional rendering of the "RESCHEDULE_ALL_HEARINGS" button based on user type and role.

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js [hunk 1]
```diff
@@ -60,6 +60,7 @@ function CaseFileAdmission({ t, path }) {
   const [updateCounter, setUpdateCounter] = useState(0);
   const roles = Digit.UserService.getUser()?.info?.roles;
   const isCaseApprover = roles.some((role) => role.code === "CASE_APPROVER");
+  const isCourtRoomManager = roles.find((role) => role.code === "COURT_ROOM_MANAGER");
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js:63]: done

**Review by anirudh-0 — APPROVED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/EvidenceModal.js [hunk 2]
```diff
@@ -824,7 +825,7 @@ const EvidenceModal = ({ caseData, documentSubmission = [], setShow, userRoles,
               </div>
             </div>
             {(userRoles.includes("SUBMISSION_RESPONDER") || userRoles.includes("JUDGE_ROLE")) && (
-              <div className="application-comment">
+              <div className={`application-comment ${isCourtRoomManager && "disabled"}`}>
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/EvidenceModal.js:828]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/components/HearingOverlayDropdown.js [hunk 3]
```diff
@@ -9,8 +9,15 @@ const OverlayDropdown = ({ styles, textStyle, column, row, master, module }) =>
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
   const history = useHistory();
   const dropdownRef = useRef(null);
+  const roles = Digit.UserService.getUser()?.info?.roles;
+  const isCourtRoomManager = roles.some((role) => role.code === "COURT_ROOM_MANAGER");
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/components/HearingOverlayDropdown.js:13]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/components/HearingOverlayDropdown.js [hunk 4]
```diff
@@ -9,8 +9,15 @@ const OverlayDropdown = ({ styles, textStyle, column, row, master, module }) =>
...
 
-  const dropdownItems = Digit.Customizations[master]?.[module]?.dropDownItems?.(row) || [];
+  const dropdownItems =
+    Digit.Customizations[master]?.[module]?.dropDownItems?.(row).filter((item) => {
+      if (isCourtRoomManager && item?.id === "view_case") {
+        return true;
+      } else return false;
+    }) || [];
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/components/HearingOverlayDropdown.js:20]: to be done later

**Pull request merged**